### ISSUE TITLE
using webpackPrefetch or webpackPreload in dynamic import() is breaking "dynamic" part of it

### ISSUE THREAD

**Issue by keblodev** [webpack/webpack#7410]
## Feature request

please consider following example:

```return foo ? import(/* webpackPrefetch: true */ './moduleA') : import(/* webpackPrefetch: true */ './moduleB');```

**What is the expected behavior?**

I would expect based on the app condition webpack to add scripts to the ```<head>``` dynamically with ```rel="prefetch"```

**What is the current behavior?**

Currently I see both scrips being added to the ```<head>``` right away on the initial page load.
To be fair the scripts are evaluated based on condition, but they still are fetched on the initial load, which goes against dynamic code splitting approach and it eats client's bandwidth.

This seems to happen only with webpackPrefetch and webpackPreload. Other options work fine.

**What is motivation or use case for adding/changing the behavior?**

In my mind webpackPrefetch and webpackPreload should not break the dynamic code splitting, which they do right now.

bellow example will load everything as expected dynamically when needed, but without ```rel="prefetch"``` property:

```return foo ? import('./moduleA') : import('./moduleB');```

**How should this be implemented in your opinion?**

webpackPrefetch and webpackPreload should just add proper ```rel``` property to the ```<link>``` 

**Are you willing to work on this yourself?**
yes

**Comment by ronanamsterdam** [comment]
@ooflorent will take a look at possible approach this/next week. Unless there's already ongoing WIP

**Comment by sokra** [comment]
This can't be "fixed".

**Comment by ronanamsterdam** [comment]
@charliedavison nope -> I just stopped using this feature. The only impact for me using `import` without `webpackPrefetch` is that I don't have `rel="prefetch"` for file links that I'm getting and in my case I've addressed that by having really small chunk sizes. Basically my localization files contain information needed only for a particular page, so I don't load the whole localization for the entire app. Which makes those request to be resolved almost instantly and not having `rel="prefetch"` in not impacting the performance in this case -> maybe that can work for you as well.
Without `webpackPrefetch` it should load files dynamically. 
And If it loads whole bunch of other files -> I don't think that's related to this feature. Try using `import` without magic comments -> that works for me.
